src/cc/arduino/plugins/wifi101/UpdaterImpl.java

```diff
@@ -27,65 +27,105 @@
+		fwAvailable.add(new WINCFlasher("WINC1501 Model B", "19.5.4", "firmwares/WINC1500/19.5.4/m2m_aio_3a0.bin", new ArrayList<String>(asList("Arduino/Genuino MKR1000"))));
+		fwAvailable.add(new WINCFlasher("WINC1501 Model B", "19.5.2", "firmwares/WINC1500/19.5.2/m2m_aio_3a0.bin", new ArrayList<String>(asList("Arduino/Genuino MKR1000"))));
+		fwAvailable.add(new WINCFlasher("WINC1501 Model B", "19.4.4", "firmwares/WINC1500/19.4.4/m2m_aio_3a0.bin", new ArrayList<String>(asList("Arduino/Genuino MKR1000"))));
+		fwAvailable.add(new WINCFlasher("WINC1501 Model A", "19.4.4", "firmwares/WINC1500/19.4.4/m2m_aio_2b0.bin", new ArrayList<String>(asList("Arduino/Genuino MKR1000", "Wifi 101"))));
 
-		refreshSerialPortList();
+		boardList.put("2341:8054", "NINA");
+		boardList.put("2341:8056", "NINA");
+		boardList.put("03eb:2145", "NINA");
+		boardList.put("2341:804e", "WINC1500");
 
+		for (Flasher firmware : fwAvailable)
+			getFirmwareSelector().addItem(firmware);
+		refreshSerialPortList();
 		websites.add("arduino.cc:443");
-
 		refreshCertList();
-
-		// flasher = new CLIFlasher() {
-		flasher = new JavaFlasher() {
-			@Override
-			public void progress(int progress, String text) {
-				if (text.length() > 60) {
-					text = text.substring(0, 60) + "...";
-				}
-				getUpdateProgressBar().setValue(progress);
-				getUpdateProgressBar().setStringPainted(true);
-				getUpdateProgressBar().setString(text);
-			}
-		};
 	}
 
-	private List<String> websites = new ArrayList<>();
-
 	private void refreshCertList() {
 		CertificateListModel model = new CertificateListModel(websites);
 		getCertSelector().setModel(model);
 	}
 
-	private SerialPortListModel listModel;
-
 	@Override
 	protected void refreshSerialPortList() {
-		listModel = new SerialPortListModel();
-		getSerialPortList().setModel(listModel);
+	DefaultListModel<String> model = new DefaultListModel<String>();
+	BoardPort board;
+	listModel = new SerialPortListModel();
+	for (int i = 0; i < listModel.getSize(); i++) {
+		board = listModel.getPort(i);
+		if (board.getBoardName() != null) {
+			model.addElement(board.getBoardName().concat(" (").concat(board.getAddress()).concat(")"));
+		} else {
+			model.addElement(board.getAddress());
+		}
+	}
+	getSerialPortList().removeAll();
+	getSerialPortList().setModel(model);
+}
+
+  @Override
+  protected	void SelectBoardModule() {
+		int added = 0;
+		getFirmwareSelector().removeAllItems();
+		String selectedBoard = boardList.get((getSelectedPort().getVID().toLowerCase()).concat(":").concat(getSelectedPort().getPID().toLowerCase()));
+		for (Flasher firmware : fwAvailable) {
+			if (firmware.getName().equals(selectedBoard) && firmware.isCompatible(getSelectedPort().getBoardName())) {
+				getFirmwareSelector().addItem(firmware);
+				added++;
+			}
+		}
+		if (added == 0) {
+			for (Flasher firmware : fwAvailable)
+				getFirmwareSelector().addItem(firmware);
+		}
+		if (selectedBoard.equals("NINA")) {
```

Comment:
Can we add something to the interface to see if it supports certs instead of hard coding?